classes/submission/reviewAssignment/Collector.php

```diff
@@ -435,8 +445,25 @@ public function getQueryBuilder(): Builder
 
         $q->when(
             $this->reviewerIds !== null,
-            fn (Builder $q) =>
-            $q->whereIn('ra.reviewer_id', $this->reviewerIds)
+            fn (Builder $q) => $q
+                ->whereIn('ra.reviewer_id', $this->reviewerIds)
+                ->when(
+                    $this->isLastReviewRoundByReviewer,
+                    fn (Builder $q) => $q
+                        // Aggregate the latest review round number for the given assignment and reviewer
+                        ->leftJoinSub(
+                            DB::table('review_assignments as ramax')
+                                ->select(['ramax.submission_id', 'ramax.reviewer_id'])
+                                ->selectRaw('MAX(ramax.round) as latest_round')
+                                ->selectRaw('MAX(ramax.stage_id) as latest_stage')
+                                ->whereIn('ramax.reviewer_id', $this->reviewerIds)
+                                ->groupBy(['ramax.submission_id', 'ramax.reviewer_id']),
+                            'agrmax',
+                            fn (JoinClause $join) => $join->on('ra.submission_id', '=', 'agrmax.submission_id')
+                        )
+                        ->whereColumn('ra.round', 'agrmax.latest_round') // assignments from the last review round per reviewer only
+                        ->whereColumn('ra.stage_id', 'agrmax.latest_stage') // assignments for the current review stage only (for OMP)
```

Comment:
Sorry to be pain.. but I think this still suffers with the edge case, if you have 2 rounds in internal and 1 round in external. You would end up with latest_round:2 and latest_stage: 3, but the latest review assignment would be having round 1 and stage 3.

Reply:
Sorry, it shows the latest in the internal stage or external in this scenario?

Reply:
My point is that the round is calculated independently from the stage.

So if there are 3 rounds in internal stage and 2 rounds in the external stage - it will be trying to find review assignment which is in 3rd round and external stage, which does not exist. Or do I miss something